### ISSUE TITLE
`deno lsp` stops working

### ISSUE THREAD

**Issue by fannheyward** [fannheyward/coc-deno#139]
@fannheyward definition jumping is now possible, but there may be other problems. The situation is such that the deno lsp environment is broken after the definition jump. :(

## Repro

1. definition jump
2. `CTRL-O` to return to the previous buffer

## Capture

https://user-images.githubusercontent.com/188642/146353966-2d6b4995-28a2-4013-a60f-f7d6570b7879.mp4

_Originally posted by @yaegassy in https://github.com/fannheyward/coc-deno/issues/135#issuecomment-995634535_

**Comment by fannheyward** [comment]
Tested with Deno 1.7.2, still have this issue. A weird behavior, you have `Deno.args` and do jumping on `args` and return back, `deno lsp` still works. But for `console.log`, after you do jumping and return back, `deno lsp` stops working.

**Comment by yaegassy** [comment]
I don't know how to adjust for this problem, but I tried comparing `coc-deno` and `vscode_deno`.

There was a difference in the results of the `deno.status` command.

In the case of `coc-deno`, it seems that the language server recognizes the virtual document as a project file :(

**coc-deno's deno.status**:

> :CocCommand deno.status

```
// ...snip
## Workspace Details

  - <details><summary>Documents in memory: 2</summary>

    - file:///Users/yaegassy/sandbox/coc-deno-check/deno:/asset/lib.deno.console.d.ts
    - file:///Users/yaegassy/sandbox/coc-deno-check/sample.ts

  </details>
// ...snip
```

**vscode_deno's deno.status (capture)**:

<img width="978" alt="vscode_deno-deno-staus" src="https://user-images.githubusercontent.com/188642/148316948-b8b89898-d7bc-4432-9bdd-8bd2ea49f19c.png">

**Comment by fannheyward** [comment]
Yes, virtual document is the root case, maybe e7195b87c29d79b7f000b00797631260ddd53c5a is not a good fix, will look into this today.

**Comment by fannheyward** [comment]
```
console.log('deno');
```

Test 1:

1. put your cursor on `console`, do definition jumping and return back with `c-o`
2. `deno lsp` works
3. put your cursor on `log`, same actions, everything works as expected

`deno.status`:

```
...
    - file:///Users/fannheyward/src/hello-deno/deno:/asset/lib.deno.console.d.ts
    - file:///Users/fannheyward/src/hello-deno/deno:/asset/lib.deno.shared_globals.d.ts
    - file:///Users/fannheyward/src/hello-deno/main.ts
...
```

Test 2:

1. put your cursor on `log`, do definition jumping and return back
2. `deno lsp` is broken
3. jumping on `console` doesn't work

`deno.status`:

```
...
    - file:///Users/fannheyward/src/hello-deno/deno:/asset/lib.deno.console.d.ts
    - file:///Users/fannheyward/src/hello-deno/main.ts
...
```

**Comment by fannheyward** [comment]
@yaegassy make a fix in v3.9.3, please give a try.

**Comment by yaegassy** [comment]
I think it's a great idea. I have confirmed that this issue has been resolved in my environment.
